### DISCUSSION TITLE
[Question] <Input> `addonBefore`  break `align-items:baseline` ?

### DISCUSSION THREAD

**Discussion by wving5** [ant-design/ant-design#51781] [Q&A]
Demo: https://stackblitz.com/edit/react-k1fmw4?file=demo.js

Any clean way to center align the entire row **without hardcoding** `minHeight: 32` on icon ?

<img width="660" alt="image" src="https://github.com/user-attachments/assets/12030ef1-9432-4fda-8e3c-d0a3bd9dad8c">

**Comment by petercat-assistant** [comment]
在使用 Ant Design 的 `<Input>` 组件时，如果 `addonBefore` 属性导致 `align-items: baseline` 的布局问题，可以尝试以下方法来居中对齐整个行，而不需要硬编码 `minHeight: 32`：

1. **使用 Flexbox 调整对齐方式**：
   - 你可以在外层容器上应用 `display: flex` 和 `align-items: center` 来确保内容在垂直方向上居中对齐。

2. **调整样式**：
   - 如果你有自定义的样式文件，可以通过调整 CSS 来实现居中对齐。例如，确保图标和输入框的高度一致，或者使用 `line-height` 来调整。

3. **使用 Ant Design 的样式覆盖**：
   - Ant Design 允许通过覆盖默认样式来实现自定义布局。你可以在项目的样式文件中覆盖相关的样式规则。

如果这些方法仍然无法解决问题，建议查看 Ant Design 的文档或社区讨论，可能会有其他开发者提供的解决方案。

如果需要进一步的帮助，请在评论中描述你的问题，并 @petercat-assistant 以开始与我对话。
